### PR TITLE
[various] Align Flutter and Dart SDK constraints

### PR DESCRIPTION
**Pull request by stuartmorgan**
As described in https://github.com/flutter/flutter/issues/121684, we currently have inconsistencies between Flutter SDK constraints and Dart SDK constraints; we have often updated only the former. This PR:
1. Adds CI enforcement via the repo tooling that the minimum versions are consistent.
2. Adds a new repo tooling command to update SDK constraints, to help mass-fix all the violations of the new enforcement in step 1 (and for future mass changes, such as when we update our test matrix and mass-drop support for versions that are no longe tested).
    - In all cases, the looser constraint was updated to match the more restrictive constraint, such that there's no actual change in what Flutter version any package actually supports.
3. Runs `dart fix --apply` over all changed packages to automatically fix all of the analysis failures caused by step 2 suddenly making all of our packages able to use `super` parameters.

Fixes https://github.com/flutter/flutter/issues/121684
Fixes https://github.com/flutter/flutter/issues/121685

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
go_router part LGTM

**Comment by stuartmorgan** [comment]
I'm overriding the version check here even though policy would normally be to release it due to the code changes; since the changes are a) automated, b) small, and c) uninteresting, I think it's better to just let these changes go out organically. Changes of them causing a regression that would impact a later release seem *very* small.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by tarrinneal — APPROVED** [review]

File: packages/cross_file/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/cross_file/CHANGELOG.md:3]: The pubspec of this package didn't change.

File: packages/flutter_markdown/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/flutter_markdown/CHANGELOG.md:3]: Nothing has changed in this package, it seems.

File: packages/flutter_migrate/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/flutter_migrate/CHANGELOG.md:3]: No changes on this package.

File: packages/flutter_template_images/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ## NEXT
 
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/flutter_template_images/CHANGELOG.md:3]: Unchanged?

File: packages/google_identity_services_web/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/google_identity_services_web/CHANGELOG.md:3]: Unchanged?

File: packages/metrics_center/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/metrics_center/CHANGELOG.md:3]: Unchanged?

File: packages/multicast_dns/CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/multicast_dns/CHANGELOG.md:3]: Unchanged?

File: packages/plugin_platform_interface/CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/plugin_platform_interface/CHANGELOG.md:3]: Unchanged?

File: packages/standard_message_codec/CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/standard_message_codec/CHANGELOG.md:3]: Unchanged?

File: packages/url_launcher/url_launcher_ios/CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/url_launcher/url_launcher_ios/CHANGELOG.md:3]: Unchanged?

File: packages/web_benchmarks/CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/web_benchmarks/CHANGELOG.md:3]: Unchanged?

File: packages/webview_flutter/webview_flutter/CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/webview_flutter/webview_flutter/CHANGELOG.md:3]: Unchanged?

File: packages/xdg_directories/CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/xdg_directories/CHANGELOG.md:3]: Unchanged?

File: third_party/packages/cupertino_icons/CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,6 @@
 ## NEXT
 
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline third_party/packages/cupertino_icons/CHANGELOG.md:3]: Unchanged?

**Review by ditman — COMMENTED** [review]
There's a few packages where only the CHANGELOG.md seems to have been updated, but not the pubspec.yaml or any other file within them?

(PS: this PR is making my browser sound like a jet taking off. I'm sorry if I flagged the wrong package in the PR, the files would jump around between I clicked and the review dialog showed up!)

**Comment by stuartmorgan** [comment]
> There's a few packages where only the CHANGELOG.md seems to have been updated, but not the pubspec.yaml or any other file within them?

Good eyes! I must have messed up the diff base for the mass update the last time I re-applied them.

And I was so close to it being landable :(

**Comment by stuartmorgan** [comment]
Oh, I see; this has bit me before, but I forgot: `--run-on-changed-packages` doesn't actually do what I want here because I changed the script, which forces all packages to be consider changed.

**Comment by ditman** [comment]
> And I was so close to it being landable :(

Sorry! (>_<) all I wanted was to RSLGTM this. Blame it on @reidbaker who made me sensitive to these things :P

**Comment by stuartmorgan** [comment]
You scored 100% by the way; re-running it with the right targeting touched 14 fewer packages.

File: packages/flutter_markdown/CHANGELOG.md (hunk 2 above)
Comment (ditman) [reply packages/flutter_markdown/CHANGELOG.md:3]: This was a lie, I meant to touch "flutter_lints"!

File: packages/flutter_lints/CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Aligns Dart and Flutter SDK constraints.
```
Comment (ditman) [inline packages/flutter_lints/CHANGELOG.md]: 🙈

Comment (ditman) [reply packages/flutter_lints/CHANGELOG.md]: (also `third_party/packages/cupertino_icons`)

**Comment by stuartmorgan** [comment]
Hm, it's failing the maps renderer selection test almost all of the time. I wonder if there's some server-side issue here where we're getting the legacy mode regardless of the requested mode in many cases?

**Comment by stuartmorgan** [comment]
Since the failure here was one of the random ARM macOS timeouts, and the same shard passed on the previous run, I'm just going to land this now before this can bitrot again, since it's such a fragile PR given the number of packages whose changelogs are touched.

**Pull request merged**